python-sdk/pachyderm_sdk/config.py

```diff
@@ -12,20 +14,65 @@
 class ConfigFile:
     """A parsed Pachyderm config file."""
 
-    def __init__(self, config_file: Union[Path, str]):
+    def __init__(self, data: dict):
         """
+        Parameters
+        ----------
+        data : dict (JSON)
+            The config file data. The only required fields are:
+            {
+              "v2": {
+                "active_context": ...,
+                "contexts": {...}
+              }
+            }
+        """
+        if (v2 := data.get("v2")) is None or not (
+            "active_context" in v2 and "contexts" in v2
+        ):
+            raise ValueError(
+                'Expected format: {"v2": {"active_context": ..., "contexts": {...}}'
+            )
+        self._config_file_data = data
+
+    @classmethod
+    def from_file(cls, config_file: Union[Path, str]) -> "ConfigFile":
```

Comment:
nit: maybe call this `from_path`? when i read `file` in Python i htink of `file` objects

Reply:
good suggestion!